pandas/core/groupby/grouper.py

```diff
@@ -509,37 +510,24 @@ def __init__(
             if self.grouper is None and self.name is not None and self.obj is not None:
                 self.grouper = self.obj[self.name]
 
+                if self.grouper.ndim > 1:
+                    # i.e. DataFrame case reachable if columns non-unique
+                    t = self.name or str(type(self.grouper))
+                    raise ValueError(f"Grouper for '{t}' not 1-dimensional")
+
             elif isinstance(self.grouper, (list, tuple)):
                 self.grouper = com.asarray_tuplesafe(self.grouper)
 
             # a passed Categorical
             elif is_categorical_dtype(self.grouper):
+                self._passed_categorical = True
```

Comment:
why don't you just make this a property of the grouper itself?

Reply:
i think what you're suggesting is something like

```
@property
def _passed_categorical(self):
    return is_categorical_dtype(self.grouper)
```

that is the longer-term goal, but ATM we treat directly-passed Categoricals differently from Categoricals derived within `__init__`, so that property wouldn't be _quite_ the same as this variable.  I expect that the different treatment is non-intentional, but need to track that down to be sure.